recipes/dns_config.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: dns_config
+#
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+# Utility function to retrieve the assigned hostname from DynamoDB
+def retrieve_assigned_hostname
+  instance_id = shell_out!('curl -s http://169.254.169.254/latest/meta-data/instance-id', user: 'root').stdout.strip
+  Chef::Log.info("Instance id is '#{instance_id}'")
+
+  assigned_hostname = shell_out!("#{node['cfncluster']['cookbook_virtualenv_path']}/bin/aws dynamodb " \
```

Comment:
this can be a problem for runtime baked AMIs where cookbook virtualenv is not available

Reply:
Interesting comment, so it seems we have a problem in multiple places:
https://github.com/aws/aws-parallelcluster-cookbook/blob/wip/hit/recipes/_master_slurm_config.rb#L63
https://github.com/aws/aws-parallelcluster-cookbook/blob/wip/hit/recipes/_master_slurm_config.rb#L78
https://github.com/aws/aws-parallelcluster-cookbook/blob/wip/hit/recipes/_master_slurm_finalize.rb#L20
https://github.com/aws/aws-parallelcluster-cookbook/blob/wip/hit/recipes/fsx_mount.rb#L35
https://github.com/aws/aws-parallelcluster-cookbook/blob/wip/hit/recipes/setup_raid_on_master.rb#L42
https://github.com/aws/aws-parallelcluster-cookbook/blob/wip/hit/recipes/slurm_install.rb#L55

right?

Reply:
Fixed by moving _setup_python.rb recipe in prep_env. To be tested with not backed AMIs